onnxruntime/core/optimizer/nchwc_transformer.cc

```diff
@@ -951,13 +952,15 @@ void NchwcTransformerImpl::TransformResize(Node& node) {
     }
   }
 
-  NodeArg* scales_arg;
+  NodeArg* sizes_arg = nullptr;
+  NodeArg* scales_arg = nullptr;
+
   if (node.SinceVersion() >= 11) {
-    // Bail out if Resize has the optional "sizes" tensor.
-    if (input_defs.size() == 3) {
+    if (input_defs.size() >= 4) {
+      sizes_arg = input_defs[3];
+    }
+    if (sizes_arg == nullptr) {
       scales_arg = input_defs[2];
```

Comment:
scales is also optional. There needs a check of >=3 here?

Reply:
Just checked the spec. no need of check if sizes_arg is null.

---
In reply to: [610788953](https://github.com/microsoft/onnxruntime/pull/7290#discussion_r610788953) [](ancestors = 610788953)